### DISCUSSION TITLE
How to create a new filtered collection when the source collection's property changed?

### DISCUSSION THREAD

**Discussion by Snailya** [reactivemarbles/DynamicData#872] [Q&A] [answered]
I'm writing a canExecute observable using reactive ui and dynamic data.
In my view, I have a ListBox with checkable item. I want to observe on this checkbox and set the canExecute to true is any of the item is checked. So i convert the itemssource for the listbox named Items to IChangeSet by ToObservableChangeSet(). I've noticed that it will notified only by calling Items.ToObservableChangeSet().WhenPropertyChanged(x=>x.IsChecked). But I don't know how to create a new filtered output collection whenever it notifies.

**Comment by JakenVeina** [comment, marked as answer]
So, you need an `IObservable<bool>` that is `true` when there is at least one item in the set that is selected?

Pull the `DynamicData.Aggregations` namespace and do...
```cs
items
    .FilterOnObservable(static item => item.WhenPropertyChanged(x => x.IsChecked))
    .IsNotEmpty();
```

**Reply by Snailya** [reply]
Thanks
